openpype/hosts/fusion/api/lib.py

```diff
@@ -333,16 +334,47 @@ def get_frame_path(path):
     return filename, padding, ext
 
 
-def get_fusion_module():
-    """Get current Fusion instance"""
-    fusion = getattr(sys.modules["__main__"], "fusion", None)
-    return fusion
+def _load_fusuionscriptlib():
+    try:
+        import fusionscript as script_module
+
+        return script_module
+    except ImportError as e:
+        lib_path = ""
+        if sys.platform.startswith("darwin"):
+            lib_path = "/Applications/Blackmagic Fusion 18/Fusion.app/Contents/MacOS/fusionscript.so"
+        elif sys.platform.startswith("win"):
+            lib_path = "C:\\Program Files\\Blackmagic Design\\DaVinci Resolve\\fusionscript.dll"
+        elif sys.platform.startswith("linux"):
+            lib_path = "/opt/BlackmagicDesign/Fusion18/fusionscript.so"
+
+        if not os.path.isfile(lib_path):
+            print("[Fusion] [Library Does Not Exist on Disk]", lib_path)
+
+        bmd = imp.load_dynamic("fusionscript", lib_path)
+
+        if not bmd:
+            raise ImportError(
+                "[BMD] Could not locate module dependencies"
+            ) from e
```

Comment:
Should we cache this - so that every call to `_load_fusuionscriptlib` and thus to `get_bmd_library` doesn't need to resolve the paths and load dynamically each time and re-use it once?

Reply:
It is "cached" as it tries to import the script module under Try. If it can't import it then it tries to import the dll file.

Reply:
You are absolutely right. 👍